core/trino-parser/src/main/java/io/trino/sql/parser/AstBuilder.java

```diff
@@ -1555,30 +1558,89 @@ public Node visitDropRole(SqlBaseParser.DropRoleContext context)
                 visitIfPresent(context.catalog, Identifier.class));
     }
 
+    @Override
+    public Node visitGrantPrivileges(SqlBaseParser.GrantPrivilegesContext context)
+    {
+        List<String> privileges;
+        if (context.ALL() != null) {
+            privileges = ImmutableList.of();
+        }
+        else {
+            privileges = extractPrivileges(context.privilegeOrRole());
+        }
+        return new Grant(
+                getLocation(context),
+                privileges.isEmpty() ? Optional.empty() : Optional.of(privileges),
+                createGrantObject(getLocation(context), context.grantObject()),
+                getPrincipalSpecification(context.principal()),
+                context.OPTION() != null);
+    }
+
     @Override
     public Node visitGrantRoles(SqlBaseParser.GrantRolesContext context)
     {
+        // Check legality for a role grant.
```

Comment:
Is this a statement about what this code is doing? If so, I don't see it. Or is it a "TO DO"?

Reply:
It was a leftover from the generic case.  Removed.